cmd/libs/go2idl/types/types.go

```diff
@@ -146,16 +187,37 @@ func (u Universe) Package(packagePath string) *Package {
 		return p
 	}
 	p := &Package{
-		Path:    packagePath,
-		Types:   map[string]*Type{},
-		Imports: map[string]*Package{},
+		Path:      packagePath,
+		Types:     map[string]*Type{},
+		Functions: map[string]*Type{},
+		Variables: map[string]*Type{},
+		Imports:   map[string]*Package{},
 	}
 	u[packagePath] = p
 	return p
 }
 
+// Nature can either be "function", "variable", or "type".
+type Nature string
+
+const (
+	Function Nature = "function"
+	Variable Nature = "variable"
+	// TODO: rename it to TYPE after we rename Type to something else.
+	TypeNature Nature = "type"
+)
+
+// TODO: "Type" is not an accurate name anymore, maybe "Object"?
 // Type represents a subset of possible go types.
 type Type struct {
+	// Nature indicates what is hold by the Object.
+	// 1. When Nature == Function, Name.Name is the function name, Signature is the
+	// function's signature;
+	// 2. When Nature == Variable, Name.Name is the variable name. (We probably needs
+	// to add a field `TYPE *Type` to denote the variable type);
+	// 3. When Nature == Type, all fields are what their comments suggest.
+	Nature Nature
```

Comment:
OK, let's chat about this tomorrow. I'm now thinking it'd be better not to change Type like this and instead, to represent functions, variables, and consts as:

```go
type Decl struct {
  Name Name
  Type *Type
}
```

The Signature type should be updated to use this for function parameters. Might have to add an "untyped" builtin when we start handling consts.